### PR TITLE
CVSB-1241

### PR DESCRIPTION
**Pull request by gmlazu**
#### CVSB-1241
* Added docker compose & dockerfile
* Added report generation for ATF reports
* Added lambda invocation to test-results
* Added S3 upload

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+# this is the name and version displayed in the SonarQube UI. Was mandatory prior to SonarQube 6.1.
+sonar.projectName=Report Generation Microservice
+sonar.projectVersion=1.0
+
+# Path is relative to the sonar-project.properties file. Replace “\” by “/” on Windows.
+# This property is optional if sonar.modules is set.
+sonar.sources=src
+
+sonar.exclusions=.build, src/models/**, src/config/**, src/models/**, src/services/LambdaService.ts, src/services/S3BucketService.ts, src/handler.ts, src/functions/**
```
Comment (acozianu) [inline sonar-project.properties]: functions should not be excluded

File: src/models/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    endpoint: string;
+}
+
+interface IActivity {
+    id: string;
+    activityType: "visit" | "wait";
+    testStationName: string;
+    testStationPNumber: string;
+    testStationEmail: string;
+    testStationType: "atf" | "gvts" | "hq";
```
Comment (acozianu) [inline src/models/index.d.ts]: please add the strings in an enum file

File: src/services/ReportGenerationService.ts [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+                        certificateNumber: reportSheet.getCell(`J${25 + k}`),
+                        expiryDate: reportSheet.getCell(`K${25 + k}`)
+                    };
+                })
+            };
+
+            reportSheet.getCell("C20").value = {
+                richText: [
+                    { font: { bold: true }, text: "Data Protection" },
+                    { text: "\nThe information provided on this form will be used for the purposes of DVSA statutory functions. " },
```
Comment (acozianu) [inline src/services/ReportGenerationService.ts]: move these strings in the enum file

File: sonar-project.properties (hunk 1 above)
Comment (gmlazu) [reply sonar-project.properties]: The functions cannot be unit tested, and do not contain core logic. All core logic has been tested

File: src/services/ReportGenerationService.ts (hunk 3 above)
Comment (gmlazu) [reply src/services/ReportGenerationService.ts]: what are you talking about?

File: src/models/index.d.ts (hunk 2 above)
Comment (gmlazu) [reply src/models/index.d.ts]: done

File: src/services/ReportGenerationService.ts [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+                declaration.date.value = moment(activity.endTime).format("DD/MM/YYYY");
+                declaration.finishTime.value = moment(activity.endTime).format("HH:mm:ss");
+
+                // Populate activity report
+                for (let i = 0, j = 0; i < template.reportTemplate.activityDetails.length && j < testResults.length; i++, j++) {
+                    const detailsTemplate: any = template.reportTemplate.activityDetails[i];
+                    const testResult: any = testResults[j];
+                    const testType: any = testResult.testTypes;
+
+                    detailsTemplate.activity.value = (activity.activityType === "visit") ? "Test" : "Wait Time";
```
Comment (adpetrescu) [inline src/services/ReportGenerationService.ts]: can you please move the hardcoded values in enums?

**Pull request merged**